WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -68,9 +95,26 @@ class PostListMainViewModel @Inject constructor(
         _postListAction.postValue(PostListAction.NewPost(site))
     }
 
+    fun updateAuthorFilterSelection(position: Int) {
```

Comment:
We can. I thought in the last PR this was a "use your discretion" comment, but I guess I'll change it.

Reply:
In the last PR there were a lot of conversations/changes going on at the same time which made it hard to review and provide the best feedback, this is one of the reasons I suggested doing a completely fresh review.

Specifically for this, I didn't have a full understanding of our options for it and @malinajirka brought up that we can use the adapter's ids which I think is better than trusting the ordinal.

> but I guess I'll change it.

You definitely don't have to, please feel free to bring up any counterpoints for this change if you have them.

Reply:
I think I have stated my opinion on this enough times. I felt there wasn't any drawbacks, and the code was more simple. No point in continuing to restate it.

Changed in 0bbec1c54fae6b5923f6301670c777582947a165

Reply:
I looked at the previous discussion on this and in [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/9446#discussion_r268458485) you mentioned that:

> We could do away with that if we passed the id via the view model and then grab that back out from the adapter to pass back to the view model. Happy to make that change if you think it will make the code simpler. Should I make that change?

You have also mentioned that you are waiting to hear back on the minimal usage of the `ordinal` in [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/9446#issuecomment-476046236).

And then @malinajirka asked my take on it in [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/9446#discussion_r268892947):

> I'm on the edge. If we stored the ordinal into a persistance storage, I'd definitely vote for removing it and store the id instead. Since we use this just in runtime, I can't think of a scenario in which it could turn against us. Wdyt @oguzkocer ?

And I have mentioned that I'd look into this in my review [in this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/9446#discussion_r268892947):

> I think we should use the id and not the ordinal.
> Hopefully we don't have to do that and even if we do, we can do it just by passing the enum itself and not the id of it. I'll check this out in my review.

And I have checked out the code and passing the `enum` didn't seem reasonable and I think I pretty much ended on your suggestion although I have forgotten about it then.

> I think I have stated my opinion on this enough times. I felt there wasn't any drawbacks, and the code was more simple. No point in continuing to restate it.

I am sorry you feel this way. I couldn't find the comment you mentioned, but your opinion certainly matters, so if you let me know your thoughts on it or link me to that comment, I'd love to understand your take on it better. As I have mentioned before there were too many conversations in the previous PR that was not very well threaded, so it was very easy to miss things which is what I was afraid of. If I missed anything there, I am sorry.

Reply:
My stance was just that using `ordinal` would be cleaner and simpler for our code, with no drawbacks since it is only used for runtime and the `ordinal` values are guaranteed to be the same as the positions.

As @malinajirka [said]:
>Since we use this just in runtime, I can't think of a scenario in which it could turn against us.

I still think the code resolving this issue isn't very clean 0bbec1c54fae6b5923f6301670c777582947a165. However if that was the intended result then let's resolve this issue.

Reply:
I think a mild usage of `ordinal` is fine, but we should try not to rely on it. Here is what I had in mind: https://cloudup.com/co0lhIPaCW8. I also reviewed that commit to give a sense of what I am thinking.

I have to say though, this decision is probably not worth all this discussion. We can definitely go on with it if you feel it's constructive and something is coming out of it, but we can also try to wrap this up as quickly as we can whichever you prefer.

Reply:
I've addressed the individual comments on the commit. I agree that we should resolve this discussion.